api/client/lib.py

```diff
@@ -334,15 +334,7 @@ def make_key(key):
 
 
 def rank_series_by_source(access_token, api_host, series_list):
-    # We sort the internal tuple representations of the dictionaries because
-    # otherwise when we call set() we end up with duplicates if iteritems()
-    # returns a different order for the same dictionary. See test case.
-    selections_sorted = set(tuple(sorted(
-        [k_v for k_v in iter(list(single_series.items()))
-         if k_v[0] not in ('source_id', 'source_name')],
```

Comment:
Now we've got that source_id-source_name not matching behavior again. Need to filter out source_name, or else if you pass in a series object like `{..., 'source_id': 2, 'source_name': 'FAO'}` from `get_data_series()` and it'll return something like

```py
[ {..., 'source_id': 2, 'source_name': 'FAO'},
  {..., 'source_id': 14, 'source_name': 'FAO'},
  {..., 'source_id': 26, 'source_name': 'FAO'} ]
```

Reply:
^ Good thing to add a unit test for as well

Reply:
ref https://github.com/gro-intelligence/api-client/pull/125/commits/04a449523641783440035464c064034270ba9572

Reply:
Done. Cleaned up so pre-selected source is ignored since the goal is to consider all possible sources